core/src/main/java/org/jdbi/v3/tweak/CollectorFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.tweak;
+
+import java.util.stream.Collector;
+
+/**
+ * Factory for building containers of elements.
+ * The collector accepts only objects of the type of the container elements.
+ *
+ * @param <T> the type of the container elements
+ * @param <R> the type of the container
+ */
+public interface CollectorFactory<T, R> {
+
+    /**
+     * Whether the corresponding collector accepts objects of the given type.
```

Comment:
Is this "accepts objects of the given type" or more accurate as "produces objects of the given type" since it will compare against e.g. `List` or `Set`